agent/utils/src/aws.rs

```diff
@@ -271,6 +271,17 @@ where
         resources,
         "Could not convert secret-access-key to String",
     )?;
+    match aws_secret.get("session-token") {
+        Some(token) => env::set_var(
+            "AWS_SESSION_TOKEN",
+            resource_agent::provider::IntoProviderError::context(
+                String::from_utf8(token.to_owned()),
+                resources,
+                "Could not convert session-token to String",
+            )?,
+        ),
+        None => (),
+    };
```

Comment:
I think this can be better expressed as:
```suggestion
    if let Some(token) = aws_secret.get("session-token") {
        env::set_var(
            "AWS_SESSION_TOKEN",
            resource_agent::provider::IntoProviderError::context(
                String::from_utf8(token.to_owned()),
                resources,
                "Could not convert session-token to String",
            )?;
    }
```